mautic/mautic
# Fix: Empty contact field value set to null on edit

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/5117
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This bugfix/feature has been sponsored by @Webmecanik

Editing a contact sets its unset core fields to the empty string ('')
This can be problematic if there are any checks on NULL values

[//]: # ( As applicable: )
#### Steps to reproduce the bug:

1. Create a new contact setting a field like First Name to any value
2. Check the database to see the contact's unset core fields (e.g. firstname, lastname, email) set to NULL
3. Update a lead setting a field like Last Name to any value
4. Check the database to see the contact's unset text core fields (e.g. firstname, lastname, email) set to ''

#### Steps to test this PR:
1.  The same steps like reproduce jsut check if  contact field was set as null in db after edit contact

![image](https://user-images.githubusercontent.com/462477/32239311-78fcb3c4-be6a-11e7-84c5-2f068f071cae.png)
